amxmodx/string.cpp

```diff
@@ -938,9 +949,9 @@ static cell AMX_NATIVE_CALL amx_strtok(AMX *amx, cell *params)
 	{
 		if (trim && !done_flag)
 		{
-			if (isspace(string[i]))
+			if ((spaces = utf8getspaces(string + i, len) > 0))
```

Comment:
Should `len` be `len - i`? (not only here)

Reply:
`utf8iscategory` is a greedy function. It will try to match spaces as much as possible. Here we want to retrieve all the spaces all once until we hit a non-space character. We are not working with a specific maximum (apart the string itself of course), so it makes no difference.

Reply:
@Arkshine so, it can be just `SIZE_MAX`?

Reply:
You could, but I'm not sure why you would want to use some weird constant.
The documentation says _"Size of the input in bytes"_ ; so the string length is fine.

Reply:
I agree for `len - i`, the string is shifted. It would be the logical approach.
Since the string is guaranteed to be null-terminated we could use `SIZE_MAX`, but the usage here in this context is wrong.

Reply:
@Arkshine we can remove the second arg from `utf8getspaces` and just pass `SIZE_MAX` in `utf8iscategory`, it isn't wrong.

Reply:
In this context, then yes, it's okay. It simplifies the code, and since the string is always null-terminated, it should be fine.